main.go

```diff
@@ -78,6 +78,7 @@ func main() {
 		Compatibility:               cfg.Compatibility,
 		PublishInternal:             cfg.PublishInternal,
 		PublishHostIP:               cfg.PublishHostIP,
+		PublishHostExternalIP:       cfg.PublishHostExternalIP,
```

Comment:
I think this is quite confusing with `PublishHostIP`. It doesn't really distinguish what is used for what.

Maybe we can change that. It should be more understandable. 

What is the reason of having two flags instead of having one?

Reply:
Good question, so what about PublishHostIPMode or something with PreferInternal/PreferExternal values? There is quite a difference as PublishHostIP will publish hostIP of matching Pods which for example in AWS clusters is a node private IP, while the flag I've added would query the pod assigned node and use it's external IP (ElasticIP in case of AWS) if applicable.